cmd/controller-manager/main.go

```diff
@@ -46,8 +46,12 @@ func main() {
 	runtime.GOMAXPROCS(runtime.NumCPU())
 	klog.InitFlags(nil)
 
+	fs := pflag.CommandLine
 	s := options.NewServerOption()
-	s.AddFlags(pflag.CommandLine)
+
+	s.AddFlags(fs)
+	commonutil.LeaderElectionDefault(&s.LeaderElection)
+	componentbaseoptions.BindLeaderElectionFlags(&s.LeaderElection, fs)
 
```

Comment:
Just mark LockObjectNamespace deprecated is not enough, we should set `ResourceNamespace` value to LockObjectNamespace  when it's not empty: )

Reply:
> Just mark LockObjectNamespace deprecated is not enough, we should set ResourceNamespace value to LockObjectNamespace when it's not empty: )

All right！